src/JsonSchema/Constraints/FormatConstraint.php

```diff
@@ -138,7 +138,7 @@ protected function validateDateTime($datetime, $format)
         // which will fail the above string comparison because the passed
         // $datetime may be '2000-05-01T12:12:12.123Z' but format() will return
         // '2000-05-01T12:12:12.123000Z'
-        if ((strpos('u', $format) !== -1) && (intval($dt->format('u')) > 0)) {
+        if ((strpos('u', $format) !== -1) && (preg_match('/\.\d+Z/', $datetime))) {
```

Comment:
Ah yes, thanks. Also, I hadn't considered `2000-05-01T12:12:12.000000+01:00`, which this will fail on.